openl3/core.py

```diff
@@ -9,11 +9,12 @@
 from numbers import Real
 from math import ceil
 import warnings
-from .models import load_audio_embedding_model, load_image_embedding_model
+from .models import load_audio_embedding_model, load_image_embedding_model, _validate_audio_frontend
 from .openl3_exceptions import OpenL3Error
 from .openl3_warnings import OpenL3Warning
 import skimage
 from moviepy.video.io.VideoFileClip import VideoFileClip
+import librosa
```

Comment:
Good catch! Tbh I thought that openl3 was already importing librosa which is why I left it at the top level.

And I just realized that it's not in setup.py but the reason that it runs fine is that kapre is dependent on librosa.

So:
 - I'll move the librosa import inside of the frontend functions
 - (though that won't make much of a difference since kapre imports it anyways)
 - do we want to add librosa as an explicit dependency to setup.py?

Reply:
Actually, I was thinking abt this earlier. 

We should have 
```python
import skimage
from moviepy.video.io.VideoFileClip import VideoFileClip
```
be optional dependencies as well - only import if you're using the image network.

And we also have a seemingly unused import that we can probably remove
```python
import sklearn.decomposition
```